ACE/ace/SOCK_Dgram.cpp

```diff
@@ -786,21 +786,34 @@ ACE_SOCK_Dgram::make_multicast_ifaddr6 (ipv6_mreq *ret_mreq,
         ACE_OS::ace_isdigit (net_if[0]) &&
         (if_ix = ACE_OS::atoi (net_if)) > 0;
 
-      IP_ADAPTER_ADDRESSES tmp_addrs;
-      // Initial call to determine actual memory size needed
-      ULONG bufLen = 0;
-      char *buf = 0;
-      if (::GetAdaptersAddresses (AF_INET6, 0, 0, &tmp_addrs, &bufLen)
-          == ERROR_BUFFER_OVERFLOW)
+      ULONG bufLen = 15000; // Initial size as per Microsoft
```

Comment:
The previous solution to query for the size of the buffer looks more optimal

Reply:
What is the problem you have that require these changes?

Reply:
I am not sure. Microsoft says this call is expensive, and 15000 guesstimate is a good start.

Reply:
What's more important, is that the initial size obtained by the first call, in some cases is still insufficient for the next call. The example suggested in the article is what Microsoft recommends.

Reply:
Ok, so it is a small performance optimization, not a real fix, code change is from https://docs.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getadaptersaddresses?

Reply:
It is a real fix. `GetAdaptersAddresses` was failing on the second trial with `ERROR_BUFFER_OVERFLOW`.

Reply:
Ok

Reply:
I was able to capture the exact behavior before and after the fix.